docs/madr/decisions/075-migrating-to-kri-based-envoy-resource-and-stat-naming.md

```diff
@@ -0,0 +1,193 @@
+# Migrating to KRI-based Envoy resource and stat naming
+
+* Status: accepted
+
+## Context and problem statement
+
+As described in the [Resource Identifier MADR](070-resource-identifier.md), we chose to standardize resource naming using the KRI (Kuma Resource Identifier) format. This change improves consistency across the system and allows for stronger cross-resource references, making data easier to present and understand. However, moving to KRI will break existing setups that rely on current Envoy stat and resource names, such as dashboards and GUI features. A smooth migration path is needed to let users adopt KRI without disruption.
+
+### Affected systems and users
+
+Two main groups are impacted:
+
+* Internal systems: Kuma GUI
+* External users: Dashboards, Prometheus
+
+#### Kuma GUI
+
+The Dataplane view displays inbound and outbound endpoints using data from Envoy stats. It matches these stats to configuration by parsing stat names, such as cluster or listener names. Changes to stat naming or resource identifiers will break this matching logic. We must coordinate changes with the GUI to support both the old and new formats during the transition.
```

Comment:
This will soon be moved to new Inspect API where all inbounds/outbounds will be accessible via `_layout` endpoint. That will provide KRI for each of these

Reply:
Yep sounds good. So we are essentially having the same view implemented using different HTTP APIs/Stats depending on the feature flag of the Dataplane/Zone Proxy.